### PR TITLE
feat: Add dark mode and lightbox for docs

### LINKED ISSUE #2005: [FR] Zoom for images in the documentation
Some images in the documentation have very small text, which is hard to read on the page. Zooming in the page doesn't help, as images are scaled to fit the window and don't actually get bigger.

It would be nice to be able to click on an image and have it pop out into an overlay with pan-and-zoom functionality. This would make it a bit more convenient than RMB > Open Image in new tab.

### PR DESCRIPTION
**Pull request by zusorio**
- Add lightbox for images
- Add dark mode
- Add SPA mode and link prefetching

Closes #2005

### CODE REVIEW

File: docs/mkdocs.yml [hunk 1]
```diff
@@ -10,6 +10,23 @@ theme:
...
+    - media: "(prefers-color-scheme: light)"
+      scheme: default
+      toggle:
+        icon: material/brightness-7
+        name: Switch to dark mode
+    - media: "(prefers-color-scheme: dark)"
+      scheme: slate
+      toggle:
+        icon: material/brightness-4
+        name: Switch to system preference
```
[the remark this answers is not in the document]
Comment (zusorio) [reply docs/mkdocs.yml:29]: Hmmm, this was taken straight from the docs but I'll take a look

[the remark this answers is not in the document]
Comment (zusorio) [reply docs/mkdocs.yml:29]: When I'm in auto mode and press the button it switches to light mode. So I think that's the correct behavior?

[the remark this answers is not in the document]
Comment (MoritzWeber0) [reply docs/mkdocs.yml:29]: Works with the latest version of mkdocs-material. I was using an outdated version.

**Review by MoritzWeber0 — APPROVED** [review]
The SPA mode only works in the development environment due to a missing `site_url`, but that's fine for now.

**Pull request merged**